src/data/zmod/quotient.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+import data.zmod.basic
+import group_theory.quotient_group
+import ring_theory.int.basic
+
+/-!
+# `zmod n` and quotient groups / rings
+
+This file relates `zmod n` to the quotient group
+`quotient_add_group.quotient (add_subgroup.gmultiples n)` and to the quotient ring
+`(ideal.span {n}).quotient`.
+
+## Main definitions
+
+ - `zmod.quotient_gmultiples_nat_equiv_zmod` and `zmod.quotient_gmultiples_equiv_zmod`:
+   `zmod n` is the group quotient of `ℤ` by `n ℤ := add_subgroup.gmultiples (n)`,
+   (where `n : ℕ` and `n : ℤ` respectively)
+ - `zmod.quotient_span_nat_equiv_zmod` and `zmod.quotient_span_equiv_zmod`:
+   `zmod n` is the ring quotient of `ℤ` by `n ℤ : ideal.span {n}`
+   (where `n : ℕ` and `n : ℤ` respectively)
+ - `zmod.lift n f` is the map from `zmod n` induced by `f : ℤ →+ A` that maps `n` to `0`.
+
+## Tags
+
+zmod, quotient group, quotient ring, ideal quotient
+-/
+
+section lift
+
+open quotient_add_group
+open zmod
+
+variables (n : ℕ) {A R : Type*} [add_group A] [ring R]
+
+namespace int
+
+/-- `ℤ` modulo multiples of `n : ℕ` is `zmod n`. -/
+def quotient_gmultiples_nat_equiv_zmod :
+  quotient (add_subgroup.gmultiples (n : ℤ)) ≃+ zmod n :=
```

Comment:
Is it possible to get this isomorphism from `quotient_ker_equiv_of_surjective`, the first isomorphism theorem, applied to the homomorphism `int.cast_add_hom` from `ℤ` to `zmod n`,

Reply:
this basically comes down to how easy it is to show the (obvious) `(gmultiples_hom (zmod n) 1).ker = add_subgroup.gmultiples n`

Reply:
`quotient_ker_equiv_of_right_inverse` would be better than `quotient_ker_equiv_of_surjective` as it's computable.

Reply:
I don't think there is one for `add_equiv`s

Reply:
```lean
noncomputable example : quotient (add_subgroup.gmultiples (n : ℤ)) ≃+ zmod n :=
begin
  suffices : (int.cast_add_hom (zmod n)).ker = add_subgroup.gmultiples ↑n,
  rw ←this, exact quotient_ker_equiv_of_surjective (int.cast_add_hom (zmod n)) int_cast_surjective,
  ext,
  simp only [int_coe_zmod_eq_zero_iff_dvd, add_monoid_hom.mem_ker,
             int.mem_gmultiples_iff, int.coe_cast_add_hom]
end
```

Heather, your `int.cast_add_hom` idea ended up working better

Reply:
(this is two lemmata really)

Reply:
> I don't think there is one for add_equivs

Indeed, there does not seem to be one for group quotients at all, only ideal quotients. I'll make a new PR.

Reply:
@ericrbg This is nice!  I feel like
```lean
(int.cast_add_hom (zmod n)).ker = add_subgroup.gmultiples ↑n
```
should be a lemma in mathlib anyway.

Reply:
> (Don't we have it? @adamtopaz said so.)

There is nothing I can find by `grep`ping for `cast_add_hom`, in any case.